src/Gir/Marshal/SymbolTable.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Gir
+{
+	public class SymbolTable : IEnumerable<ISymbol>
+	{
+		readonly Dictionary<string, ISymbol> typeMap = new Dictionary<string, ISymbol>();
+
+		public SymbolTable()
+		{
+			// Add primitives
+			AddType(new Primitive("void", "void", String.Empty));
+			AddType(new Primitive("gpointer", "IntPtr", "IntPtr.Zero"));
+			AddType(new Primitive("gboolean", "bool", "false"));
+			AddType(new Primitive("gint", "int", "0"));
+			AddType(new Primitive("guint", "uint", "0"));
+			AddType(new Primitive("int", "int", "0"));
+			AddType(new Primitive("unsigned", "uint", "0"));
+			AddType(new Primitive("unsigned int", "uint", "0"));
+			AddType(new Primitive("unsigned-int", "uint", "0"));
+			AddType(new Primitive("gshort", "short", "0"));
+			AddType(new Primitive("gushort", "ushort", "0"));
+			AddType(new Primitive("short", "short", "0"));
+			AddType(new Primitive("guchar", "byte", "0"));
+			AddType(new Primitive("unsigned char", "byte", "0"));
+			AddType(new Primitive("unsigned-char", "byte", "0"));
+			AddType(new Primitive("guint1", "bool", "false"));
+			AddType(new Primitive("uint1", "bool", "false"));
+			AddType(new Primitive("gint8", "sbyte", "0"));
+			AddType(new Primitive("guint8", "byte", "0"));
+			AddType(new Primitive("gint16", "short", "0"));
+			AddType(new Primitive("guint16", "ushort", "0"));
+			AddType(new Primitive("gint32", "int", "0"));
+			AddType(new Primitive("guint32", "uint", "0"));
+			AddType(new Primitive("gint64", "long", "0"));
+			AddType(new Primitive("guint64", "ulong", "0"));
+			AddType(new Primitive("long long", "long", "0"));
+			AddType(new Primitive("gfloat", "float", "0.0"));
+			AddType(new Primitive("float", "float", "0.0"));
+			AddType(new Primitive("gdouble", "double", "0.0"));
+			AddType(new Primitive("double", "double", "0.0"));
+			AddType(new Primitive("goffset", "long", "0"));
+			AddType(new Primitive("GQuark", "int", "0"));
+		}
+
+		public void AddType (ISymbol symbol)
+		{
+			typeMap[symbol.CType] = symbol;
+		}
+
+		public void ProcessAliases ()
```

Comment:
Do we want to fully resolve the aliases? It might make sense to actually generate classes for aliases. On the native side it might not make a difference to pass in a guint8 or GDateDay but I think this matters in C#. If we have a type for GDateDay then one can methods for creating a GDateDay or extracting information from it. If it's a plain guint8 then this is not possible.

Reply:
For example, in GLib there's an alias for GDateYear which maps to guint16. But there are also a lot of functions that take or return GDateYear. If we create classes for GDateYear then we can express that certain functions take a year and not a primitive int. This is more type safe and also more clear since the intention is in the type.